satpy/tests/reader_tests/test_fci_l2_nc.py

```diff
@@ -74,124 +96,124 @@ def setUp(self):
         }
         self.assertEqual(global_attributes, expected_global_attributes)
 
-    @mock.patch('satpy.readers.fci_l2_nc.get_area_definition')
+    @mock.patch('satpy.readers.fci_l2_nc.geometry.AreaDefinition')
     @mock.patch('satpy.readers.fci_l2_nc.make_ext')
     def test_area_definition(self, me_, gad_):
         """Test the area definition computation."""
-        self.reader._compute_area_def()
+        self.fh._compute_area_def(make_dataid(name='test_area_def', resolution=2000))
 
         # Asserts that the make_ext function was called with the correct arguments
         me_.assert_called_once()
-        name, args, kwargs = me_.mock_calls[0]
-        self.assertTrue(np.allclose(args[0], 0.0))
-        self.assertTrue(np.allclose(args[1], 515.6620))
-        self.assertTrue(np.allclose(args[2], 0.0))
-        self.assertTrue(np.allclose(args[3], 5672.28217))
-        self.assertTrue(np.allclose(args[4], 35786400.))
-
-        p_dict = {
-            'nlines': 100,
-            'ncols': 10,
-            'ssp_lon': 10.0,
-            'a': 6378137.,
-            'b': 6356752.,
-            'h': 35786400.,
-            'a_name': 'FCI Area',
-            'a_desc': 'Area for FCI instrument',
-            'p_id': 'geos'
-        }
+        args, kwargs = me_.call_args
+        self.assertTrue(np.allclose(args, [-0.0, -515.6620, 5672.28217, 0.0, 35786400.]))
+
+        proj_dict = {'a': 6378137.,
+                     'lon_0': 0.0,
+                     'h': 35786400,
+                     "rf": 298.257223563,
+                     'proj': 'geos',
+                     'units': 'm',
+                     'sweep': 'y'}
 
         # Asserts that the get_area_definition function was called with the correct arguments
         gad_.assert_called_once()
-        name, args, kwargs = gad_.mock_calls[0]
-        self.assertEqual(args[0], p_dict)
-        # The second argument must be the return result of the make_ext function
-        self.assertEqual(args[1]._extract_mock_name(), 'make_ext()')
+        args, kwargs = gad_.call_args
+        self.assertEqual(args[0], 'mtg_fci_fdss_2km')
+        self.assertEqual(args[1], 'MTG FCI Full Disk Scanning Service area definition with 2 km resolution')
+        self.assertEqual(args[2], '')
+        self.assertEqual(args[3], proj_dict)
+        self.assertEqual(args[4], 10)
+        self.assertEqual(args[5], 100)
 
     def test_dataset(self):
         """Test the execution of the get_dataset function."""
         # Checks the correct execution of the get_dataset function with a valid file_key
-        dataset = self.reader.get_dataset(None,
-                                          {'file_key': 'test_one_layer',
-                                           'fill_value': -999, 'mask_value': 0.,
-                                           'file_type': 'test_file_type'})
+        dataset = self.fh.get_dataset(make_dataid(name='test_one_layer', resolution=2000),
```

Comment:
I think this function could be split into multiple smaller ones, according to the comments